### PR TITLE
Add custom upgrade subscription page layout

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

This PR adds a custom layout to the upgrade subscription page.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203623179643290/1203906467538586/f) _(internal)_

## 🔍 What does this change?

- adds support for a `getLayout` function on next pages

## 🛡 What tests cover this?

Manual testing.

## ❓ How to test this?

Go to the upgrade subscription page, and observe the new layout (locally or in the preview deployment)

## 📹 Demo

image

### CODE REVIEW

File: apps/site/src/pages/settings/billing/upgrade/upgrade-page-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          borderTopWidth: 1,
+          borderTopStyle: "solid",
+          borderTopColor: ({ palette }) => palette.gray[20],
+        }}
+      >
+        <Container>
+          <BlockProtocolIcon
+            sx={{ color: ({ palette }) => palette.gray[40], height: 24 }}
+          />
+        </Container>
```
Comment (nonparibus) [inline apps/site/src/pages/settings/billing/upgrade/upgrade-page-layout.tsx:47]: The logo is larger than it should be. Please style-match Figma.

File: apps/site/src/components/pages/auth-wall.tsx [hunk 2]
```diff
@@ -1,15 +1,14 @@
-import { NextPage } from "next";
 import { useRouter } from "next/router";
-import { FunctionComponent } from "react";
```
Comment (nonparibus) [inline apps/site/src/components/pages/auth-wall.tsx:2]: @benwerner01 Minor tweak while we're working on this page: please can we increase the font-weight of the `PLANNED` labels next to certain features, everywhere they appear on the site (including here on the upgrade checkout pages?)

They're not meant to be as thin as they're appearing. See Figma. Cheers!

File: apps/site/src/pages/settings/billing/upgrade/upgrade-page-layout.tsx (hunk 1 above)
Comment (benwerner01) [reply apps/site/src/pages/settings/billing/upgrade/upgrade-page-layout.tsx:47]: Which logo, the top logo or the one in the footer?

Comment (benwerner01) [reply apps/site/src/pages/settings/billing/upgrade/upgrade-page-layout.tsx]: I've adjusted both in https://github.com/blockprotocol/blockprotocol/pull/1152/commits/c8a63b7e89f065f4d03d1638c7765b40d0dd0521, let me know if that's closer to what you had in mind

File: apps/site/src/components/pages/auth-wall.tsx (hunk 2 above)
Comment (benwerner01) [reply apps/site/src/components/pages/auth-wall.tsx:2]: Adjusted these in https://github.com/blockprotocol/blockprotocol/pull/1152/commits/0f190302c26add3eda765e288693aabaaa314418

image

**Comment by nonparibus** [comment]
Happy with functionality/visuals. Over to code review. @blockprotocol/frontend

**Pull request merged**